### PR TITLE
Add paystar gateway

### PR DESCRIPTION
**Pull request by majidfeiz**
Added PayStar driver
Added PayStar config file
Update documentations

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
         }
     ],
     "require": {
-        "php": "^8.1",
+        "php": "^8.0|^8.1",
```
Comment (omalizadeh) [inline composer.json:37]: Minimum version for this branch of package is 8.1 and i don't want to decrease it.

File: config/gateway_paystar.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+     *  driver class namespace.
+     */
+    'driver' => Omalizadeh\MultiPayment\Drivers\Paystar\Paystar::class,
+
+    /**
+     *  gateway configurations.
+     */
+    'main' => [
+        'pin' => '',
+        'secret' => '',
```
Comment (omalizadeh) [inline config/gateway_paystar.php:15]: where is 'secret' used?

File: config/gateway_paystar.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     *  driver class namespace.
+     */
+    'driver' => Omalizadeh\MultiPayment\Drivers\Paystar\Paystar::class,
+
+    /**
+     *  gateway configurations.
+     */
+    'main' => [
+        'pin' => '',
```
Comment (omalizadeh) [inline config/gateway_paystar.php:14]: it was not 'pin' in paystar config. it was something like gateway_id?

File: src/Drivers/Paystar/Paystar.php [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            $mobile = $this->checkPhoneNumberFormat($mobile);
+        }
+
+        $callback=$this->getInvoice()->getCallbackUrl() ?: $this->settings['callback'];
+        return [
+            'amount' => $this->getInvoice()->getAmount(),
+            'callback' => $callback,
+            'mobile' => $mobile,
+            'order_id' => $this->getInvoice()->getInvoiceId(),
+            'description' => $description,
```
Comment (omalizadeh) [inline src/Drivers/Paystar/Paystar.php:91]: where is 'sign' key from documentation?

File: src/Drivers/Paystar/Paystar.php [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        ];
+    }
+
+    protected function getVerificationData(): array
+    {
+
+        return [
+            'ref_num'=>$this->getInvoice()->getTransactionId(),
+            'amount'=>$this->getInvoice()->getAmount()
```
Comment (omalizadeh) [inline src/Drivers/Paystar/Paystar.php:100]: again no sign

File: src/Drivers/Paystar/Paystar.php [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        ];
+    }
+
+    protected function getStatusMessage(int|string $statusCode): string
+    {
+        $messages = [
```
Comment (omalizadeh) [inline src/Drivers/Paystar/Paystar.php:106]: it's not like the status table in the documentation!

File: src/Drivers/Paystar/Paystar.php [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    {
+        return 'https://core.paystar.ir/api/pardakht/verify';
+    }
+
+    private function getRequestHeaders(): array
+    {
+        return [
+            'Content-Type' => 'application/json',
+            'Accept' => 'application/json',
+            'Authorization'=> 'Bearer '.$this->settings['pin']
```
Comment (omalizadeh) [inline src/Drivers/Paystar/Paystar.php:148]: you can use withToken on Http facade instead of passing header

**Review by omalizadeh — CHANGES_REQUESTED** [review]
Check paystar doc carefully

**Comment by majidfeiz** [comment]
- [X] secret use for generate sign but sign is not required it's optional and now user can use if need.
- [X] changed pin to gateway_id 
- [X] complete status array
- [X] used withToken on Http facade instead of passing header
- [X] changed PHP version to 8.1

File: composer.json (hunk 1 above)
Comment (omalizadeh) [reply composer.json:37]: this is not changed @majidfeiz

File: config/gateway_paystar.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+     * gateway_id is required
+     * type is required => direct | pardakht
+     * use_sign is optional => true | false
+     * secret_key is optional => If use sign is true fill this value, It's your gateway secret key for generate sign
+     */
+    'main' => [
+        'gateway_id' => '',
+        'secret_key' => '',
+        'type' => '',
+        'use_sign' => '',
```
Comment (omalizadeh) [inline config/gateway_paystar.php:21]: if it's bool why you wrote quotes? set a default value

File: src/Drivers/Paystar/Paystar.php (hunk 6 above)
Comment (omalizadeh) [reply src/Drivers/Paystar/Paystar.php:138]: still got problem in messages. -1xx codes are NOT from paystar

**Review by omalizadeh — CHANGES_REQUESTED** [review]
Check comments

File: config/gateway_paystar.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     *  driver class namespace.
+     */
+    'driver' => Omalizadeh\MultiPayment\Drivers\Paystar\Paystar::class,
+
+    /**
+     *  gateway configurations.
+     * gateway_id is required
+     * type is required => direct | pardakht
+     * use_sign is optional => true | false
```
[the remark this answers is not in the document]
Comment (majidfeiz) [reply config/gateway_paystar.php:14]: It's not required. They made a mistake in documentation. I called the support team and talked with a backend team in Paystar.

File: config/gateway_paystar.php [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     *  driver class namespace.
+     */
+    'driver' => Omalizadeh\MultiPayment\Drivers\Paystar\Paystar::class,
+
+    /**
+     *  gateway configurations.
+     * gateway_id is required
+     * type is required => direct | pardakht
```
[the remark this answers is not in the document]
Comment (majidfeiz) [reply config/gateway_paystar.php:13]: It has two gateways for payments and the API base url is different

https://docs.paystar.ir/docs/tutorial-basics/IPGD
https://docs.paystar.ir/docs/tutorial-basics/IPG

File: src/Drivers/Paystar/Paystar.php (hunk 6 above)
Comment (majidfeiz) [reply src/Drivers/Paystar/Paystar.php:138]: It's here :

https://docs.paystar.ir/docs/tutorial-basics/IPGD

File: config/gateway_paystar.php (hunk 2 above)
Comment (majidfeiz) [reply config/gateway_paystar.php]: It is used to generate a sign key

File: config/gateway_paystar.php (hunk 8 above)
Comment (majidfeiz) [reply config/gateway_paystar.php:21]: Some times developers set secret_key value but they don't need to use sign because it's optional

Comment (omalizadeh) [reply config/gateway_paystar.php:21]: when they don't need to use sign they set it to false. use_sign is bool, isn't it? i asked why the quotes?

Comment (majidfeiz) [reply config/gateway_paystar.php]: Done

File: config/gateway_paystar.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (majidfeiz) [reply config/gateway_paystar.php:13]: I don't know but Paystar has 2 gateways for transactions and this driver can support both of them.

**Comment by omalizadeh** [comment]
OK, thanks for PR. please fix errors on action (styles) then we can merge it. @majidfeiz

**Comment by majidfeiz** [comment]
I fixed the code style, please merge again, @omalizadeh.

**Comment by omalizadeh** [comment]
Still not passing style check. use php cs fixer to automatically fix styles and commit it @majidfeiz

**Review by omalizadeh — APPROVED** [review]

**Pull request merged**